src/Http/Routing/src/ParameterPolicyActivator.cs

```diff
@@ -113,30 +113,24 @@ private static IParameterPolicy CreateParameterPolicy(IServiceProvider? serviceP
             // arguments that can be resolved from DI
             //
             // For example, ctor(string, IService) will beat ctor(string)
-            var matchingConstructors = constructors
-                .Where(ci => GetNonConvertableParameterTypeCount(serviceProvider, ci.GetParameters()) == arguments.Length)
-                .OrderByDescending(ci => ci.GetParameters().Length)
-                .ToArray();
+            var matchingConstructors = GetMatchingConstructors(constructors, serviceProvider, arguments.Length);
```

Comment:
This Linq removal is a bit lower priority. I think it only saves ~4KB in NativeAOT size and is a bit more code.

However, in the canonical case of only a single ctor on the Policy class, it is less allocations. So I felt it was worth it. But if someone feels strongly that we should keep the original code, this one could be reverted.

Reply:
What's the code coverage around this? Are there tests with different number of policy ctors to hit each path?

Reply:
Doing a quick check, I think the code coverage looks good. Here are some tests:

https://github.com/dotnet/aspnetcore/blob/e9863d38cce6e56b38931db8e0f1eb3176d054ad/src/Http/Routing/test/UnitTests/DefaultParameterPolicyFactoryTest.cs#L432-L509

I've added one more test here.